Automatic Request for open365-base-with-dependencies-docker

vHanda opened the pull request.

Cotes committed:
Modify cups to create PDF file in /tmp folder instead of user folder

User folder does not have write permission.

Cotes committed:
Add bus master user and password to /tmp/global.env

This is necessary to fix printing bug. When we send the notification to
the client we need to use bus credentials.

afiestas commented on run.sh:
```diff
@@ -106,6 +106,8 @@ service cups restart
 echo "export EYEOS_UNIX_USER=$EYEOS_UNIX_USER" > /tmp/global.env
 echo "export BUS_ADDRESS_HOST=$BUS_ADDRESS_HOST" >> /tmp/global.env
 echo "export BUS_SUBSCRIPTION=$BUS_SUBSCRIPTION" >> /tmp/global.env
+echo "export EYEOS_BUS_MASTER_PASSWD=$EYEOS_BUS_MASTER_PASSWD" >> /tmp/global.env
```
Just to make damn sure, this master password is in reality the user card, isn't it?

kampde replied:
yes, it is the user card. I'd like someone to fix this before it gets merged because it is wildly misleading.

Commit:
compact check for launching autosave

Commit:
use already defined variable

Commit:
autosave libreoffice documents depending on envar value

We are not autosaving if not told explicitly.

Commit:
print output of autosave.py script

jcarlosn committed:
Added deferred fps to 30 to improve performance under certain scenarios

Cotes committed:
Remove recent file list from libreoffice.

Files in recent list from libreoffice freezes GUI due to a webdav
problem.

Cotes committed:
Revert "Remove recent file list from libreoffice."

This reverts commit edc13855df36a44356479282ed6dd5ae67bf3d04.

Cotes committed:
Remove recent file list from libreoffice.

Instead of removing the file list when libreoffice closes, we remove it
at the startup.

vHanda committed:
OfficeClipboard: Re-connect each time

1. Use a local pipe instead of a TCP connection. We do not need the
extra overhead.

2. Reconnect each time. This is so that we can avoid a bug. Currently
whenever anyone tries to open a word document when LO is running (via
File -> Open) LO may crash. It crashes while trying to process an URP
request (Uno remote protocol). This script is what causes it. The script
also crashes with a "URP Bridge Disconnected" error.

There does not seem to be a convenient way to figure out if we can or
cannot make URP requests from the script. The only way seems to be to
try a fresh connection each time. It's a workaround, but it was either
this or that we increase the interval between our requests and hope for
the best. (Or fix LO)

Commit:
Use the webdav mountpoint as home to launch libreoffice

while bind-mounts are disabled, we tell libreoffice that our home is the
webdav mount point is our home so when opening files it directly lists
our libraries.

Commit:
Run a save libreoffice script when client disconnected

Also run the office scripts under the correct user

Commit:
Delete LO config

We downgraded the LO version to 5.0 from 5.1, and the configuration
makes certain documents break and we get incorrect window sizes.
(Hypothesies)

This is a hack and workaround.

jordins committed:
Run migrations if exists when running container

We need to do some migrations for existing users, so we've created a
very simple way of doing it.

jordins committed:
Copy correctly skel files to config

It was just copying files and not copying folders (like libreoffice).

jordins committed:
Move libreoffice migration to office-docker migrations

This is done now inside open365-office-docker repo

Commit:
add .dockerignore

Using this our images will be thinner because they won't add those
files/folders.

afiestas committed:
Set the delay upload to 5seconds

This will basically cache the open +w operation that libreoffice is
performing 4 times.

afiestas committed:
Set davfs dir_refresh to 1h

This cache is only for the file metadata which means that operations
open (either file or directory) will not hit the cache.

This will improve alot the performance because libreoffice performs lots
of accesss/lstat operations.

afiestas committed:
Set davfs file_refresh cache to 5 seconds

This basically batches open operations during an amount of time. Again
really useful because libreoffice would open the same file many times
(with the same flags btw) resulting in multiple file downloads.

We want this to be a really low cache value because otherwise the app
will be desync from seafile resulting in a bad user experience.

afiestas committed:
Enable davfs gui_optimize

This basically makes davfs use less amount of PROPFIND operations by
batching them. As the name of the option indicates it is thought to
improve the performance of GUI applications like libreoffice.

Commit:
Check if replacement file exists

If libreoffice hasn't been opened and the replacement is attempted, it fails and crashes

Cotes committed:
Sets delay_upload to 0 seconds

If delay_upload is equal to 5 seconds (or different than 0), docx files
weren't being saved.

vHanda committed:
Do a chown on the libreoffice config

Sometimes the libreoffice config seems to be owned by root. We aren't
sure why. This is a temporary quick fix.

kampde commented on run.sh:
```diff
@@ -79,6 +79,8 @@ if [[ ! -f /home/$SPICE_USER/.eyeosConfigured ]]; then
 	touch /home/$SPICE_USER/.eyeosConfigured
 fi
 
+chown -R 1000:1000 /home/$SPICE_USER/.config/libreoffice
```
This recursive `chown` (and others present in `run.sh`) are problematic because of latency of the dav mount (files are thousands of kms away from the container). It is possible to avoid `chown`ing anything?
